lametro/events.py

```diff
@@ -501,83 +505,66 @@ def find_approved_minutes(self, event):
-                "January 27, 2022": "Regular Board Meeting MINUTES - January 27, 2022",
-                "February 24, 2022": "MINUTES - February 24, 2022 RBM",
-                "June 23, 2022": "Regular Board Meeting MINUTES - June 23, 2022",
-                "December 1, 2022": "Regular Board Meeting MINUTES - December 1, 2022",
-            }
-
-            if date in handled_cases:
-                attachment_name = handled_cases[date]
-                (attachment,) = [
-                    each
-                    for each in attachments
-                    if each["MatterAttachmentName"] == attachment_name
-                ]
-                return attachment
+            attachments = self.get(attachment_url).json()
 
+            if len(attachments) == 0:
+                raise ValueError("No attachments for the approved minutes matter")
+            elif len(attachments) == 1:
+                valid_attachments.append(attachments)
             else:
-                try:
-                    (attachment,) = [
-                        each
-                        for each in attachments
-                        if "minutes" in each["MatterAttachmentName"].lower()
-                    ]
-                except ValueError:
-                    LOGGER.critical(
-                        "More than one attachment for the approved minutes matter"
-                    )
-                else:
-                    msg = (
-                        "More than attachment for minutes matter {0}, using {1}".format(
-                            matter["MatterId"], attachment["MatterAttachmentName"]
-                        )
-                    )
-                    self.info(msg)
-                    return attachment
+                """
+                Multiple attachments have been found.
+                Return only those that look like minutes files.
+                """
+                for attach in attachments:
+                    url = attach["MatterAttachmentHyperlink"]
+                    response = requests.get(url)
+
+                    with io.BytesIO(response.content) as filestream:
+                        pdf = pdfplumber.open(filestream)
+                        cover_page = pdf.pages[0]
+
+                        cover_page_text = cover_page.extract_text()
+                        if not cover_page_text:
+                            # No extractable text found.
+                            # Turn the page into an image and use OCR to get text.
+                            pdf_image = cover_page.to_image(resolution=150)
+
+                            with io.BytesIO() as in_mem_image:
+                                pdf_image.save(in_mem_image)
+                                in_mem_image.seek(0)
+                                cover_page_text = pytesseract.image_to_string(Image.open(in_mem_image))
+
+                    if "MINUTES" in cover_page_text:
```

Comment:
Is capitalization meaningful? If not, I'd recommend coercing `cover_page_text` to one case, to avoid missing "minutes", "Minutes", etc.
```suggestion
                    if "MINUTES" in cover_page_text.upper():
```